frontend/src/components/homepage/Filter.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+
+import ToggleButton from "../navigation/ToggleButton";
+import "./Filter.css";
+
+export type FilterProps = {
+  approvedLanguages: { [name: string]: number };
+  level: number;
+  setLevel: (newState: number) => void;
+  language: string;
+  setLanguage: (newState: string) => void;
+  role: boolean;
+  setRole: (newState: boolean) => void;
+};
+
+const Filter = ({
+  approvedLanguages,
+  level,
+  setLevel,
+  language,
+  setLanguage,
+  role,
+  setRole,
+}: FilterProps) => {
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setLanguage(event.target.value);
+  };
+  const handleRangeChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setLevel(parseInt(event.target.value, 10));
+  };
+  const languageOptions = Object.keys(approvedLanguages).map((lang) => (
+    <option value={lang}>{lang}</option>
+  ));
+  const maxLvl = approvedLanguages[language];
+  const lvlOptions = [];
+  for (let i = 1; i <= maxLvl; i += 1) {
+    lvlOptions.push(<option value={i.toString()} label={i.toString()} />);
+  }
+  return (
+    <div className="filter">
+      <h1>Filter Stories</h1>
+      <div>
+        <h4>TRANSLATION LANGUAGE</h4>
+        <select
+          name="language"
+          id="language"
+          value={language}
+          onChange={handleSelectChange}
+        >
+          {languageOptions}
+        </select>
+      </div>
+      <div>
+        <h4>ROLE</h4>
+        <ToggleButton
```

Comment:
not sure if this is relevant to the comment Emma made about the css for the Role button but when i view the site on my laptop instead of monitor i see this (could also just be my laptop being weird sorry):
![image](https://user-images.githubusercontent.com/32009013/123529801-07ee7680-d6c2-11eb-8101-abf03345d26f.png)

Reply:
LOL mine's also not having it, but in a different way

![image](https://user-images.githubusercontent.com/40974372/123530473-bdbcc380-d6c8-11eb-9816-adffc3d8fd76.png)